packages/common/src/hardhat/tasks/compile.js

```diff
@@ -0,0 +1,33 @@
+const path = require("path");
+
+// This file is mostly taken from the modified `compile` task file written by Synthetix: https://github.com/Synthetixio/synthetix
+
+const { internalTask } = require("hardhat/config");
+const { TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS } = require("hardhat/builtin-tasks/task-names");
+
+const CONTRACTS_DIR = path.resolve(__dirname, "../../../../core/contracts");
+
+// This overrides a buidler internal task, which is part of its compile task's lifecycle.
+// This allows us to filter on whitelisted, OVM-compatible contracts from the compilation list,
+// which are entries in extended `compileWhitelist` configuration.
+internalTask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS, async (_, { config, network }, runSuper) => {
```

Comment:
Is this supposed to run just for optimism compilations or will this run for all of our compilations? If just optimism, where is that specified?

Reply:
I see no reason why this couldn't be used for any network, which is how its currently configured.

Reply:
Gotcha, makes sense.